### PR TITLE
Close bad connections to takers

### LINKED ISSUE #303: Don't open p2p port via browser
If one connects to the maker's p2p port via the browser one will crash the node. 

The maker will print the following and becomes unresponsive. 

```
2021-10-14 11:17:18  WARN Error while receiving message from taker 145d3ae8-a8b9-4763-9acc-d60692d7ad35: frame size too big
2021-10-14 11:17:18  WARN Error while receiving message from taker fc3909fc-6df2-46a6-822c-97b13f999543: frame size too big
2021-10-14 11:17:18  WARN Error while receiving message from taker fc3909fc-6df2-46a6-822c-97b13f999543: frame size too big
2021-10-14 11:17:18  WARN Error while receiving message from taker 145d3ae8-a8b9-4763-9acc-d60692d7ad35: frame size too big
2021-10-14 11:17:18  WARN Error while receiving message from taker 145d3ae8-a8b9-4763-9acc-d60692d7ad35: frame size too big
```

### PR DESCRIPTION
**Pull request by thomaseizinger**
- Introduce dedicated actor for forwarding only `Ok` messages
- Close connections with takers that don't adhere to the message format

### CODE REVIEW

**Review by da-kami — APPROVED** [review]
nifty, but a bit hard to understand

File: daemon/src/forward_only_ok.rs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+{
+    async fn handle(
+        &mut self,
+        Message(result): Message<TOk, TErr>,
+        _: &mut xtra::Context<Self>,
+    ) -> KeepRunning {
+        let ok = match result {
+            Ok(ok) => ok,
+            Err(e) => {
+                tracing::error!("Stopping forwarding due to error: {}", e);
```
Comment (bonomat) [inline daemon/src/forward_only_ok.rs:42]: Are you happy with losing the cause? 

```suggestion
                tracing::error!("Stopping forwarding due to error: {:#}", e);
```

**Review by bonomat — APPROVED** [review]
Nice, thanks.

**Review by luckysori — APPROVED** [review]

File: daemon/src/forward_only_ok.rs (hunk 1 above)
Comment (thomaseizinger) [reply daemon/src/forward_only_ok.rs:42]: I don't mind using alternate display, will not make a difference for most error types. Rendering causes of errors using the _alternate_ sigil is a feature of `anyhow::Error`, not of the `std::error::Error` type.

So unless `TErr` is `anyhow::Error`, it won't make a difference.

**Pull request merged**